pkg/testmachinery/testflow/validation.go

```diff
@@ -99,6 +107,55 @@ func Validate(identifier string, tf tmv1beta1.TestFlow, locs locations.Locations
 	return nil
 }
 
+func ensureArtifactsFromStepsExist(identifier string, tf tmv1beta1.TestFlow) error {
+	for i, step := range tf {
+		identifier := fmt.Sprintf("%s.[%d].artifactsFrom", identifier, i)
+		if step.ArtifactsFrom != "" {
+			if !previousStepExists(step, step.ArtifactsFrom, tf) {
+				return fmt.Errorf("%s; Invalid value: Couldn't find artifactsFrom step %s", identifier, step.ArtifactsFrom)
+			}
+		}
+	}
+	return nil
+}
+
+func getStep(tf tmv1beta1.TestFlow, needle string) *tmv1beta1.DAGStep {
```

Comment:
this can be removed if you add `usedStepNames` to the parameters of `ensureArtifactsFromStepsExist` and `previousStepExists`.

Then you simply pass a pointer around and can get the step as before with `usedStepNames[dependentStepName] `